cmd/juju/commands/upgrademodel.go

```diff
@@ -206,7 +206,8 @@ func (c *baseUpgradeCommand) precheckVersion(ctx *cmd.Context, agentVersion vers
 		// Only upgrade to a different major number if:
 		// 1 - Explicitly requested with --agent-version or using --build-agent, and
 		// 2 - The model is running a valid version to upgrade from.
-		allowed, minVer, err := upgrades.UpgradeAllowed(agentVersion, c.Version)
+		// We will do this check in server side later, but it doesn't hurt to do it here as a precheck.
+		allowed, minVer, err := upgradevalidation.UpgradeToAllowed(agentVersion, c.Version)
```

Comment:
i thought we were moving this serverside

Reply:
Yeah, this should be done only if BestVersion() < 10